### DISCUSSION TITLE
Questions about AI group creation

### DISCUSSION THREAD

**Discussion by GoldJohnKing** [Vdauphin/HeartsAndMinds#1077] [Question and answer] [answered]
Hi, I'm trying to introduce Lambs Danger's reinforcement system into my modified version of H&M, and I have some questions.

- I've noticed that patrol groups spawned by H&M have a "delete group" waypoint once they arrive their final target. Lambs Danger said the reinforcement system may override waypoints. So I wonder if waypoints are overwritten, will H&M check and recreate the "delete" waypoint? If not, what would happen if the patrol units engage their enemies? Will they lasts forever?

- `btc_fnc_data_spawn_group` used to spawn groups from saved data. However I did not found anything about group's variable (which are assigned by setVariable). I need to use `_group setVariable ["lambs_danger_enableGroupReinforce", true, true];` to add Lambs Danger's reinforcement system to groups. If I don't implement it here, will group loaded from saved data lost the variable?

Thanks!

**Comment by Vdauphin** [comment, marked as answer]
Hello,

> will H&M check and recreate the "delete" waypoint?

There is no loop. Waypoint handle all the stuff for us

> f not, what would happen if the patrol units engage their enemies? Will they lasts forever?

Yes, and it is very bad because you won't have new patrol until those are deleted

 >  If I don't implement it here, will group loaded from saved data lost the variable?

Yes, spawn group create new group so if you don't had the variable here you won't see it

Cheers

**Reply by GoldJohnKing** [reply]
What about enemy patrols on vanilla H&M engaged by players? Will they also lasts forever unless being eliminated or deleted?

**Reply by Vdauphin** [reply]
> Will they also lasts forever unless being eliminated or deleted?

No because if player don't kill them, they will follow the waypoint they received

**Comment by JWB502** [comment]
Sory if I am comprehending wrong but does that mean there are adverse effects for using LAMBS Danger for H&M missions?

**Reply by Vdauphin** [reply]
Looks like I guess

**Reply by GoldJohnKing** [reply]
Yes, that's true.

Once H&M's patrol units are assigned as LAMBS Reinforcement, the original waypoint by H&M will be overwritten by LAMBS, that would cause patrol units won't reach their final destination and being deleted by H&M.

However, LAMBS Reinforcement can only be enable by manually setting variables on unit group. If you do not do that, it's okay to use LAMBS with H&M.

**Comment by Breech99** [comment]


**Reply by Vdauphin** [reply]
Hello,
No off topic please
Cheers
